frontend/src/components/question/editModal/EditQuestionModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useContext, useState } from "react";
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  Button
+} from '@chakra-ui/react';
+import { PRIMARY_COLOR } from '../../../CommonStyles';
+import ModalPage1 from '../modalPages/ModalPage1';
+import ModalPage2 from '../modalPages/ModalPage2';
+import { NewQuestionContext } from '../../../contexts/NewQuestionContext';
+import { QuestionString } from '../../../Commons';
+
+interface Props {
+  isVisible: boolean;
+  questionToEdit: QuestionString; // Added this prop to pass in the question to edit
+  closeHandler: () => void;
+  submitUpdateHandler: (updatedQuestion: QuestionString) => void;
+}
+
+const ModalButton = ({ label, color, onClick }: { label: string; color: string, onClick: () => void }) => {
+  return (
+    <Button colorScheme={ color } mr={3} onClick={onClick}>
+      {label}
+    </Button>
+  );
+};
+
+const EditQuestionModal: React.FC<Props> = ({ isVisible, questionToEdit, closeHandler, submitUpdateHandler }) => {
+  const [page, setPage] = useState(1);
+  const { setQuestionData, questionData } = useContext(NewQuestionContext);
+
+  function close() {
+    closeHandler();
+    setQuestionData(questionData);
```

Comment:
Should change to:
```suggestion
    setQuestionData(questionStringTemplate);
```
Though doing this does not reset the question data somehow:
See issue #17